apm-sniffer/apm-toolkit-activation/apm-toolkit-micrometer-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/micrometer/SpanHelper.java

```diff
@@ -39,10 +42,16 @@ static String tryToGetPeer(String remoteAddress, KeyValues allKeyValues) {
                                .orElse("unknown");
         try {
             URI uri = URI.create(result);
-            if (uri.getHost() == null) {
-                return null;
+            if (uri.getHost() != null) {
+                return uri.getHost() + ":" + uri.getPort();
+            }
+            if (!StringUtil.isEmpty(remoteName)) {
```

Comment:
Now I'm wondering if we shouldn't do the same remoteName check if there was an exception. Cause if there was an exception when doing `URI.create()` then maybe we can fallback to remote name?

Reply:
It seems reasonable.